### DISCUSSION TITLE
New features

### DISCUSSION THREAD

**Discussion by eyllanesc** [HEIGE-PCloud/DoIt#407] [Ideas]
These are some that I want to implement: 

1. [ ] Add social links and description for each author.
2. [ ] Add shortcode similar to [attachments shortcode](https://learn.netlify.app/en/shortcodes/attachments/ ) of [hugo-theme-learn](https://github.com/matcornic/hugo-theme-learn) theme.
3. [x] Add `Hosted on`.
4. [ ] Support for [Forestry.io](https://forestry.io/).
5. [ ] Support for [Netlify](https://www.netlify.com/).

@HEIGE-PCloud please give me some opinion about the features.

**Comment by HEIGE-PCloud** [comment]
> Support for Forestry.io
> Support for Netlify

@eyllanesc I have just played with Forestry.io for a while (and I assume Netlify is similar), what do we need to do in order to "support" it? I think it is a front-end editor for the content and seems to work just fine (while we have not "supported" it yet).

**Reply by eyllanesc** [reply]
In the case of foresrtry.io, it is only necessary to indicate some configuration files, similarly for Netlify.

**Comment by HEIGE-PCloud** [comment]
> Add Host on

#409 I consider this as a good first issue, so if anyone else want to contribute to this theme (for the first time), this will be a good starting point :)

**Reply by eyllanesc** [reply]
My comment is precisely to receive feedback on the functionalities that I want to implement, in the case of `host on`, what would be the correct location? Would it be in the same line as "Powered by Hugo | Theme - DoIt"?

**Reply by HEIGE-PCloud** [reply]
"Powered by Hugo | Host on GitHub Pages | Theme - DoIt" The `Host on` should support i18n and `GitHub Pages` can be changed to any HTML that can be set in `params.footer.hoston`.

**Comment by HEIGE-PCloud** [comment]
>  Add social links and description for each author

I am thinking that to what extent should we support the multiple authors function……? After all, this blog theme is originally designed for personal blog use, not for large platforms or organizations that publish posts with multiple authors all the time. My original purpose of accepting the multiple author proposal is to allow the theme to be able to deal with *some* circumstances when you write blog with one of your partners, or someone has kindly edited your post, or given some valuable advices, and you want to mention them as a co-author. But with more and more features adding to the multiple authors function, it seems to make the theme less personal. The user can surely choose not to use this feature, and these changes will not make any difference. But for people who do consider the multi author support as a necessary function, is DoIt theme, which is designed mainly for *personal* use, suits their needs (design language, content management, ...)?

Tell me what you think. Do you use the multi-author feature? If yes, what do you use the DoIt theme for (blog? or any other things?)

**Reply by eyllanesc** [reply]
I plan to create a blog, I already have several posts written but not published, but my idea is to invite other users to also publish, so I want to give it credit and one of those ways is to share the social links. I plan to implement that feature in https://hugodoit.pages.dev/authors/dillon/ since it currently supports links and mail. 

I don't want to extend the multi-user feature any further, I just want to improve what already exists.

**Reply by HEIGE-PCloud** [reply]
OK, I agree with you.

<img width="1128" alt="image" src="https://user-images.githubusercontent.com/52968553/146995166-90b69d9d-85ef-4483-8acf-b1d8828640f1.png">

I think the author page needs some redesign (just pile all social icons below the name does not look nice) and probably need to take the avatar #365 in to account as well. But simply reuse the design of the homepage does not seem appropriate.

Maybe some layout similar to the [Hugo Eureka](https://github.com/wangchucheng/hugo-eureka) theme might works well. https://www.wangchucheng.com/en/authors/wang

<img width="1128" alt="image" src="https://user-images.githubusercontent.com/52968553/146996160-d5de8bc5-0f1b-405d-9441-4ab1d509d0ad.png">

**Reply by shmishkat** [reply]
This one looks good!
